pilot/pkg/bootstrap/server.go

```diff
@@ -471,14 +473,13 @@ func (s *Server) WaitUntilCompletion() {
 }
 
 // initSDSServer starts the SDS server
-func (s *Server) initSDSServer(args *PilotArgs) {
+func (s *Server) initSDSServer() {
 	if s.kubeClient != nil {
 		if !features.EnableXDSIdentityCheck {
 			// Make sure we have security
 			log.Warnf("skipping Kubernetes credential reader; PILOT_ENABLE_XDS_IDENTITY_CHECK must be set to true for this feature.")
 		} else {
-			// TODO move this to a startup function and pass stop
```

Comment:
@howardjohn - I'm starting to have doubts about watching Secrets in all clusters for SDS. Too broad and dangerous permission needed. It should be somehow possible for user to reduce the permissions granted to istio-reader, and to have SDS only use local k8s.